internal/pppoe/pppoe.go

```diff
@@ -117,3 +124,29 @@ func (c *Conn) Close() error {
 	}
 	return nil
 }
+
+func (c *Conn) Read(b []byte) (int, error) {
+	n, _, err := readSessionPacket(c.sessionFd, b, c.readDeadline)
+	return n, err
+}
+
+func (c *Conn) Write(b []byte) (int, error) {
+	// TODO: deadline
+	return sendSessionPacket(c.sessionFd, b, c.writeDeadline)
+}
+
+func (c *Conn) SetDeadline(deadline time.Time) error {
```

Comment:
Technically these three methods don't implement the net.Conn interface rules faithfully, which require that:

```go
// SetReadDeadline sets the deadline for future Read calls
// and any currently-blocked Read call.
// A zero value for t means Read will not time out.
```

Note the "and any currently-blocked Read call" part.

net/http depends on that, for instance.

It's checked by https://godoc.org/golang.org/x/net/nettest#TestConn too.

Reply:
So maybe you just want to document these as incomplete for now in the godoc and in the body a `TODO: if anybody ever cares`.

Reply:
Done.

My expectation is that this won't get fixed until/unless the runtime's netpoller gets exposed. At which point I can just hook all of this into the main netpoller and have it correctly implement full timeout semantics for me with its more complex logic.